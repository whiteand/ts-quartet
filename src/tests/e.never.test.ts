import { e as v } from "..";
import { testValidatorImpure } from "./testValidatorImpure";

describe("v.never", () => {
  test("v.never", () => {
    testValidatorImpure(
      v(v.never),
      [],
      [null, false, [], {}, 1, 0, NaN, undefined, true]
    );
  });
});
