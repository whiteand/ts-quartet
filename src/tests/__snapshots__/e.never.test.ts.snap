// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v.never v.never 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": null,
  },
]
`;

exports[`v.never v.never 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": false,
  },
]
`;

exports[`v.never v.never 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": Array [],
  },
]
`;

exports[`v.never v.never 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": Object {},
  },
]
`;

exports[`v.never v.never 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": 1,
  },
]
`;

exports[`v.never v.never 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": 0,
  },
]
`;

exports[`v.never v.never 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": NaN,
  },
]
`;

exports[`v.never v.never 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": undefined,
  },
]
`;

exports[`v.never v.never 9`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": true,
  },
]
`;
